bin/dune_init.ml

```diff
@@ -519,7 +519,43 @@ module Component = struct
             [ File.make_text (Path.parent_exn opam_file) (Path.basename opam_file) "" ]
           | Esy -> [ File.make_text dir "package.json" "" ]
         in
-        { dir; files = dune_project_file dir opts :: package_files }
+        let default_files =
+          let fmt_version =
+            try
+              let ic = Unix.open_process_in "ocamlformat --version" in
+              let version = input_line ic in
+              let _ = Unix.close_process_in ic in
+              (let open Pp.O in
+              Console.print_user_message
+                (User_message.make
+                   [ Pp.tag User_message.Style.Hint (Pp.verbatim "Format")
+                     ++ Pp.textf ": using ocamlversion number %s" version
+                   ]));
+              version
```

Comment:
Hello! Sorry to be coming late to this PR.

The design of the `init` subcommand centers around the values of type `Component.t`, that help abstract over the kinds of stuff it will create. See

https://github.com/ocaml/dune/blob/581f852/bin/dune_init.mli#L23-L25

Rather than creating these files unconditionally, and rather than extending the project component directly, I suggest we add a new component (or two) to support this kind of tooling config file. This would offer at least the following benefits:

- We could add a subcommand (or flag) to enable creation of just these files, which could be useful as the amount of such files grows (e.g., maybe we also want to include generation of CI configs).
- It provides a natural place for providing options for the kind of customization and specialization that @toastal has been advocating for, since each component is parameterized on a set of options.
- It will keep the design of `init` more composable and easier to read about, which should help maintenance in the future: rather than creating files 

The concrete steps for this change would be:

1. Specify an aptly named component (maybe like `Tooling_configs` or do we want a different abstraction?)
2. Write a function that will take the options that configure the component to a list of [targets](https://github.com/ocaml/dune/blob/581f852/bin/dune_init.ml#L310.-L314).
3. Put those pieces together to extend the existing targets being generated by the project component in https://github.com/ocaml/dune/blob/581f852/bin/dune_init.ml#L552-L559

This feels like the right approach to me given the existing design, but I'd be willing to discuss if you (or others) feel it isn't in keeping with the aims here.

Reply:
Actually, on further thought, we can always add this on refactor. No need to complicate your PR. Sorry for the noise!